custom-elements/htmlconstructor/newtarget.html

```diff
@@ -5,100 +5,100 @@
 <script src="/resources/testharnessreport.js"></script>
 <script src="../resources/custom-elements-helpers.js"></script>
-
+<body>
 <script>
 "use strict";
 
 test_with_window(w => {
-  let afterDefinition = false;
+  let beforeDefinition = true;
   const proto1 = { "proto": "number one" };
   const proto2 = { "proto": "number two" };
 
-  const TestElement = (function () {
-    assert_throws({ name: "prototype throws" }, () => {
-      const o = Reflect.construct(w.HTMLElement, [], new.target);
-
-      assert_equals(Object.getPrototypeOf(o), proto2,
-        "Must use the value returned from new.target.prototype");
-      assert_not_equals(Object.getPrototypeOf(o), proto1,
-        "Must not use the prototype stored at definition time");
-    });
-  }).bind({});
+  function TestElement() {
+    const o = Reflect.construct(w.HTMLElement, [], new.target);
+    assert_equals(Object.getPrototypeOf(o), proto2,
+      "Must use the value returned from new.target.prototype");
+    assert_not_equals(Object.getPrototypeOf(o), proto1,
+      "Must not use the prototype stored at definition time");
+  }
 
-  Object.defineProperty(TestElement, "prototype", {
-    get() {
-      return beforeDefinition ? proto1 : proto2;
-    }
+  const ElementWithDynamicPrototype = new Proxy(TestElement, {
+      get: function (target, name) {
+        if (name == "prototype")
+          return beforeDefinition ? proto1 : proto2;
+        return target[name];
+      }
   });
 
-  w.customElements.define("test-element", TestElement);
-
-  beforeDefinition = true;
-  new TestElement();
+  w.customElements.define("test-element", ElementWithDynamicPrototype);
 
+  beforeDefinition = false;
+  new ElementWithDynamicPrototype();
 }, "Use NewTarget's prototype, not the one stored at definition time");
 
 test_with_window(w => {
   // We have to not throw during define(), but throw during super()
   let throws = false;
 
-  const TestElement = (function () {
+  function TestElement() {
+    throws = true;
     assert_throws({ name: "prototype throws" }, () => {
-      return Reflect.construct(w.HTMLElement, [], new.target);
+      Reflect.construct(w.HTMLElement, [], new.target);
     });
-  }).bind({});
+  }
 
-  Object.defineProperty(TestElement, "prototype", {
-    get() {
-      if (throws) {
+  const ElementWithDynamicPrototype = new Proxy(TestElement, {
+    get: function (target, name) {
+      if (throws && name == "prototype")
         throw { name: "prototype throws" };
-      }
-      return {};
+      return target[name];
     }
   });
 
-  w.customElements.define("test-element", TestElement);
+  w.customElements.define("test-element", ElementWithDynamicPrototype);
 
-  throws = true;
-  new TestElement();
+  new ElementWithDynamicPrototype();
 
 }, "Rethrow any exceptions thrown while getting the prototype");
 
-test_with_window(w => {
-  for (const notAnObject of [null, undefined, 5, "string"]) {
+[null, undefined, 5, "string"].forEach(function (notAnObject) {
```

Comment:
This creates duplicate test names; you'll need to change the test name to include the thing being tested.